src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/CropEdgeGeometry.cpp

```diff
@@ -266,35 +264,33 @@ Result<> CropEdgeGeometry::operator()()
     if(edgeIntersectingBoundary)
     {
       // Found an edge intersecting the boundary
+      // Calculate the interpolated value for the outside vertex and store it in the interpolatedValuesMap
+      uint64 insideVertexIdx = v0;
+      uint64 outsideVertexIdx = v1;
+      if(v1_inside)
+      {
+        insideVertexIdx = v1;
+        outsideVertexIdx = v0;
+      }
+
+      float32 insideX = srcVertices[3 * insideVertexIdx + 0];
+      float32 insideY = srcVertices[3 * insideVertexIdx + 1];
+      float32 insideZ = srcVertices[3 * insideVertexIdx + 2];
+      float32 outsideX = srcVertices[3 * outsideVertexIdx + 0];
+      float32 outsideY = srcVertices[3 * outsideVertexIdx + 1];
+      float32 outsideZ = srcVertices[3 * outsideVertexIdx + 2];
+
```

Comment:
Moving this outside of the ifs is a good way to consolidate the code. I agree with the change in behavior here, but I am just going to document it, just in case.

This does change the behavior from the original case of
```
if(behavior == BoundaryIntersectionBehavior::FilterError)
```
the shorthand if for `v1_inside` was an inversion of the behavior in
```
if(behavior == BoundaryIntersectionBehavior::InterpolateOutsideVertex)
```

That said the error case was exclusively for feedback to the user, so preserving the behavior of the actual interpolation was the right decision. It also makes more sense to expose the values actually used in the interpolation to the user for debugging incoming data.